src/imf_reader/sdr/read_announcements.py

```diff
@@ -0,0 +1,103 @@
+"""Module to get SDR data from the IMF website
+
+
+info: https://www.imf.org/en/About/Factsheets/Sheets/2023/special-drawing-rights-sdr
+
+"""
+
+from functools import lru_cache
+import pandas as pd
+import calendar
+from bs4 import BeautifulSoup
+from datetime import datetime
+
+from imf_reader.utils import make_request
+from imf_reader.config import logger, NoDataError
+
+BASE_URL = "https://www.imf.org/external/np/fin/tad/"
+MAIN_PAGE_URL = "https://www.imf.org/external/np/fin/tad/extsdr1.aspx"
+
+
+def read_tsv(url: str) -> pd.DataFrame:
+    """Read a tsv file from a url and return a dataframe"""
+
+    try:
+        return pd.read_csv(url, delimiter="/t", engine="python")
+
+    except pd.errors.ParserError:
+        raise ValueError("SDR _data not available for this date")
+
+
+def clean_df(df: pd.DataFrame) -> pd.DataFrame:
+    """Clean the SDR dataframe"""
+
+    df = df.iloc[3:, 0].str.split("\t", expand=True)
+    df.columns = ["entity", "holdings", "allocations"]
+
+    return df.assign(
+        holdings=lambda d: pd.to_numeric(
+            d.holdings.str.replace(r"[^\d.]", "", regex=True), errors="coerce"
+        ),
+        allocations=lambda d: pd.to_numeric(
+            d.allocations.str.replace(r"[^\d.]", "", regex=True), errors="coerce"
+        ),
+    ).melt(
+        id_vars="entity", value_vars=["holdings", "allocations"], var_name="indicator"
+    )
+
+
+def format_date(month: int, year: int) -> str:
+    """Return a date as year-month-day where day is the last day in the month"""
+
+    last_day = calendar.monthrange(year, month)[1]
+    return f"{year}-{month}-{last_day}"
+
+
+@lru_cache
+def get_data(year: int, month: int):
```

Comment:
Given that there are multiple functions that get data, it would be useful to give this a more meaningful name like `get_holdings_and_allocations_data`